mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,688 @@
+                h.insert(script_pubkey.clone());
+            },
+            None => {
+                let mut script_pubkeys = HashSet::new();
+                script_pubkeys.insert(script_pubkey.clone());
+                registered_txs.insert(*txid, script_pubkeys);
+            },
+        }
+    }
+
+    pub fn add_output(&self, output: WatchedOutput) {
+        let mut registered_outputs = self.registered_outputs.lock();
+        registered_outputs.push(output);
+    }
+
+    async fn check_if_tx_is_onchain(&self, txid: Txid) -> Result<bool, MmError<UtxoRpcError>> {
+        if let Err(err) = self
+            .coin
+            .as_ref()
+            .rpc_client
+            .get_transaction_bytes(&H256Json::from(txid.as_hash().into_inner()).reversed())
+            .compat()
+            .await
+            .map_err(|e| e.into_inner())
+        {
+            if let UtxoRpcError::ResponseParseError(ref json_err) = err {
+                if let JsonRpcErrorType::Response(_, json) = &json_err.error {
+                    if let Some(message) = json["message"].as_str() {
+                        if message.contains("'code': -5") {
+                            return Ok(false);
+                        }
+                    }
+                }
+            }
+            return Err(err.into());
+        }
+        Ok(true)
+    }
+
+    async fn process_tx_for_unconfirmation<T>(&self, txid: Txid, monitor: Arc<T>)
+    where
+        T: Confirm,
+    {
+        match self.check_if_tx_is_onchain(txid).await {
+            Ok(tx_is_onchain) => {
+                if !tx_is_onchain {
+                    log::info!(
+                        "Transaction {} is not found on chain. The transaction will be re-broadcasted.",
+                        txid,
+                    );
+                    monitor.transaction_unconfirmed(&txid);
+                }
+            },
+            Err(e) => log::error!(
+                "Error while trying to check if the transaction {} is discarded or not :{}",
+                txid,
+                e
+            ),
+        }
+    }
+
+    pub async fn process_txs_unconfirmations(
+        &self,
+        chain_monitor: Arc<ChainMonitor>,
+        channel_manager: Arc<ChannelManager>,
+    ) {
+        // Retrieve channel manager transaction IDs to check the chain for un-confirmations
+        let channel_manager_relevant_txids = channel_manager.get_relevant_txids();
+        for txid in channel_manager_relevant_txids {
+            self.process_tx_for_unconfirmation(txid, channel_manager.clone()).await;
+        }
+
+        // Retrieve chain monitor transaction IDs to check the chain for un-confirmations
+        let chain_monitor_relevant_txids = chain_monitor.get_relevant_txids();
+        for txid in chain_monitor_relevant_txids {
+            self.process_tx_for_unconfirmation(txid, chain_monitor.clone()).await;
+        }
+    }
+
+    async fn get_confirmed_registered_txs(
```

Comment:
It's difficult to read this function 😅 
Could you please refactor it by splitting into multiple functions?